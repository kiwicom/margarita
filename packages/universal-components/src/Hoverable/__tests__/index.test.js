// @flow

import * as React from 'react';
import { Platform, View } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';

import { Hoverable } from '..';

const originalPlatform = Platform.OS;
const onMouseEnter = jest.fn();
const onMouseLeave = jest.fn();

afterEach(() => {
  Platform.OS = originalPlatform;
});

describe('Hoverable - web', () => {
  Platform.OS = 'web';
  const { getByType } = render(
    <Hoverable onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
      <View />
    </Hoverable>,
  );

  it('should contain a children', () => {
    expect(getByType(View)).toBeDefined();
  });

  it('should execute onMouseEnter method', () => {
    fireEvent(getByType(View), 'mouseEnter');
  });

  it('should execute onMouseLeave method', () => {
    fireEvent(getByType(View), 'mouseLeave');
  });

  it('should match snapshot', () => {
    Platform.OS = 'web';
    const component = render(
      <Hoverable onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
        <View />
      </Hoverable>,
    );

    expect(component).toMatchSnapshot();
  });
});

describe('Hoverable - native', () => {
  it('should match snapshot', () => {
    Platform.OS = 'ios';
    const component = render(
      <Hoverable onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
        <View />
      </Hoverable>,
    );

    expect(component).toMatchSnapshot();
  });
});
