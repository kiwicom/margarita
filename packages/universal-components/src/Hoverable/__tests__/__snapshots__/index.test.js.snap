// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hoverable - native should match snapshot 1`] = `<View />`;

exports[`Hoverable - web should match snapshot 1`] = `
<View
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
